go-controller/pkg/clustermanager/defaultnetworkcustomization/ensurenad.go

```diff
@@ -0,0 +1,44 @@
+package defaultnetworkcustomization
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	nadtypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	nadclientset "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned"
+	nadlisters "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/listers/k8s.cni.cncf.io/v1"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+)
+
+// EnsureDefaultNetworkNAD ensure that a well-known NAD exists for the
+// default network in ovn-k namespace.
+func EnsureDefaultNetworkNAD(nadLister nadlisters.NetworkAttachmentDefinitionLister, nadClient nadclientset.Interface) (*nadtypes.NetworkAttachmentDefinition, error) {
+	nad, err := nadLister.NetworkAttachmentDefinitions(config.Kubernetes.OVNConfigNamespace).Get(types.DefaultNetworkName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+	if nad != nil {
+		return nad, nil
+	}
+	return nadClient.K8sCniCncfIoV1().NetworkAttachmentDefinitions(config.Kubernetes.OVNConfigNamespace).Create(
+		context.Background(),
+		&nadtypes.NetworkAttachmentDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      types.DefaultNetworkName,
+				Namespace: config.Kubernetes.OVNConfigNamespace,
+			},
+			Spec: nadtypes.NetworkAttachmentDefinitionSpec{
+				Config: fmt.Sprintf("{\"cniVersion\": \"0.4.0\", \"name\": \"ovn-kubernetes\", \"type\": \"%s\"}", config.CNI.Plugin),
```

Comment:
I know we are just moving code around, but now that we are a it It may be better to use the NetConf struct and marshal it, I think there are helpers around for that.